### ISSUE TITLE
Lit is in dev mode warning message shows up on localhost

### ISSUE THREAD

**Issue by mwawrusch** [lit/lit#4621]
### Which package(s) are affected?

Lit Core (lit / lit-html / lit-element / reactive-element)

### Description

I am indirectly using lit through some other package and always get the 'Lit is running in dev mode' console warning, which is rather distracting. 
It would be fantastic if you would **NOT** show this message when **window.location.hostname** is **localhost**.

### Reproduction

See above.

### Workaround

I have not found a workaround

### Is this a regression?

No or unsure. This never worked, or I haven't tried before.

### Affected versions

all

### Browser/OS/Node environment

all

**Comment by sorvell** [comment]
Yeah, that can be annoying, but I'm not sure we can say it's always right to not show the warning on localhost.

Here's a workaround. Put this script before loading anything that contains Lit:

```html
<script>
  const c = `Lit is in dev mode. Not recommended for production! See https://lit.dev/msg/dev-mode for more information.`
  window.litIssuedWarnings = new Set();
  litIssuedWarnings.add(c);
</script>
```

**Comment by justinfagnani** [comment]
If I were on a team I would want that snippet to issue it's own warning so it's not shipped to production.
